server/service/osquery_utils/queries.go

```diff
@@ -1394,6 +1398,32 @@ func sanitizeSoftware(h *fleet.Host, s *fleet.Software, logger log.Logger) {
 	}
 }
 
+// shouldRemoveSoftware returns whether or not we should remove the given Software item from this
+// host's software list.
+func shouldRemoveSoftware(h *fleet.Host, s *fleet.Software) bool {
+	checkers := []struct {
+		shouldRemove func(h *fleet.Host, s *fleet.Software) bool
+	}{
+		// Parallels is a common VM software for MacOS. Parallels makes the VM's applications
+		// visible in the host as MacOS applications, which leads to confusing output (e.g. a MacOS
+		// host reporting that it has Notepad installed when this is just an app from the Windows VM
+		// under Parallels). We want to filter out those "applications" to avoid confusion.
+		{
+			shouldRemove: func(h *fleet.Host, s *fleet.Software) bool {
+				return h.Platform == "darwin" && strings.HasPrefix(s.BundleIdentifier, "com.parallels.winapp")
+			},
+		},
+	}
+
+	for _, c := range checkers {
+		if remove := c.shouldRemove(h, s); remove {
+			return true
+		}
+	}
+
+	return false
+}
```

Comment:
I like your idea of using something different from `sanitizeSoftware`. I would advocate to start simple here and define this function as:

```go
func shouldRemoveSoftware(h *fleet.Host, s *fleet.Software) bool {
    return h.Platform == "darwin" && strings.HasPrefix(s.BundleIdentifier, "com.parallels.winapp")
}
```
if you're envisioning to add more checkers in the short term, why can we define the slice like this?

```go
checkers := shouldRemove func(h *fleet.Host, s *fleet.Software) bool
```

only bothering with this because the real behavior of this function was a bit hidden under the other boilerplate.

Reply:
> I would advocate to start simple here 

Yeah, I went back and forth on that. I don't know if there are plans to add more of these checkers, and I agree we should start things simple as a default! I think it stuck out to me as something worth making more generic since we had multiple cases within `sanitizeSoftware` and this seemed like the other side of the coin. I'm fine with making it just the basic implementation you have here, since the function signature wouldn't need to change if we _do_ end up needing more checkers.

also maybe filters is a better name?

Reply:
> making it just the basic implementation you have here, since the function signature wouldn't need to change if we do end up needing more checkers.

💯 fully agree with that!

> also maybe filters is a better name?

oh yeah, way more descriptive!